### PR TITLE
CBG-4053: bump go version to 1.22.5

### PR DESCRIPTION
**Pull request by gregns1**
CBG-4053

Bumping go version to 1.22.4

## [Integration Tests](https://jenkins.sgwdev.com/job/SyncGateway-Integration/build?delay=0sec)
- [ ] `GSI=true,xattrs=true` https://jenkins.sgwdev.com/job/SyncGateway-Integration/000/

### CODE REVIEW

File: manifest/product-config.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
             "release_name": "Couchbase Sync Gateway",
             "production": true,
             "interval": 30,
-            "go_version": "1.22.2",
+            "go_version": "1.22.4",
```
Comment (torcolvin) [inline manifest/product-config.json]: Let's change this version for 3.1.8 too in this file - this is read from `main` branch.

Comment (torcolvin) [reply manifest/product-config.json]: Oops, I mean 3.1.9

Comment (gregns1) [reply manifest/product-config.json]: Updated version in "manifest/3.1.xml" to reflect this

**Review by torcolvin — DISMISSED** [review]

**Comment by gregns1** [comment]
Will put this on ice till Tuesday for 1.22.5

**Review by torcolvin — APPROVED** [review]

**Pull request merged**